spotify/http.py

```diff
@@ -164,17 +164,20 @@ def route(
 
         method, url, = route
 
-        if self.bearer_info is None:
-            self.bearer_info = bearer_info = await self.get_bearer_info()
-            access_token = bearer_info["access_token"]
-        else:
-            access_token = self.bearer_info["access_token"]
+        headers = kwargs.pop("headers", {})
```

Comment:
This change was to avoid an infinite loop if the HTTPUserClient is constructed with token=None which happens if the user is constructed using User.from_refresh_token.